Modules/WebSocket/Sources/WebSocket/Frame.swift

```diff
@@ -134,87 +138,90 @@ struct Frame {
         if payloadLength == 126 {
-            return data.toInt(size: 2, offset: 2)
+            return data.toInt(2, offset: 2)
         } else if payloadLength == 127 {
-            return data.toInt(size: 8, offset: 2)
+            return data.toInt(8, offset: 2)
         }
         return payloadLength
     }
 
-    private var maskKey: Data {
+    fileprivate var maskKey: Buffer {
         if payloadLength <= 125 {
-            return Data(data[2 ..< 6])
+            return data.subdata(in:2 ..< 6)
         } else if payloadLength == 126 {
-            return Data(data[4 ..< 8])
+            return data.subdata(in:4 ..< 8)
         }
-        return Data(data[10 ..< 14])
+        return data.subdata(in:10 ..< 14)
     }
 
-    private var totalFrameSize: UInt64 {
+    fileprivate var totalFrameSize: UInt64 {
         let extendedPayloadExtraBytes = (payloadLength == 126 ? 2 : (payloadLength == 127 ? 8 : 0))
         let maskBytes = masked ? 4 : 0
         return UInt64(2 + extendedPayloadExtraBytes + maskBytes) + extendedPayloadLength
     }
 
-    private(set) var data = Data()
+    fileprivate(set) var data = Buffer()
 
     init() {}
 
-    init(opCode: OpCode, data: Data, maskKey: Data) {
-        self.data.append((1 << 7) | (0 << 6) | (0 << 5) | (0 << 4) | opCode.rawValue)
-
-        let masked = maskKey.count == 4
+    init(opCode: OpCode, data: Buffer, maskKey: Buffer) {
+      self.data.append([(1 << 7) | (0 << 6) | (0 << 5) | (0 << 4) | UInt8(opCode.rawValue)], count: 1)
+      
+        let masked: Bool = maskKey.count == 4
+      let mask: UInt8 = masked ? 1 : 0
         let payloadLength = UInt64(data.count)
 
         if payloadLength > UInt64(UInt16.max) {
-            self.data.append((masked ? 1 : 0) << 7 | 127)
-            self.data += Data(number: payloadLength)
+            self.data.append([mask << 7 | 127], count: 1)
+            self.data.append(Buffer(number: payloadLength))
         } else if payloadLength > 125 {
-            self.data.append((masked ? 1 : 0) << 7 | 126)
-            self.data += Data(number: UInt16(payloadLength))
+            self.data.append([mask << 7 | 126], count: 1)
+            self.data.append(Buffer(number: UInt16(payloadLength)))
         } else {
-            self.data.append((masked ? 1 : 0) << 7 | (UInt8(payloadLength) & 0x7F))
+            self.data.append([mask << 7 | (UInt8(payloadLength) & 0x7F)], count: 1)
         }
         if masked {
-            self.data += maskKey
+            self.data.append(maskKey)
 
-            var maskedData = data
+          
+          var maskedData = [UInt8](repeating: 0, count: data.count)
+          data.copyBytes(to: &maskedData, count: data.count)
 
             var maskOffset = 0
             for i in 0..<maskedData.count {
                 maskedData[i] ^= maskKey[maskOffset % 4]
                 maskOffset += 1
             }
 
-            self.data += maskedData
+          self.data.append(maskedData, count: maskedData.count)
```

Comment:
I believe this appends `maskedData` `count` amount of times, which is not what we want